dlpy/layers.py

```diff
@@ -596,6 +659,278 @@ def num_bias(self):
+    init : string, optional
+        Specifies the initialization scheme for the layer.
+        Valid Values: XAVIER, UNIFORM, NORMAL, CAUCHY, XAVIER1, XAVIER2, MSRA, MSRA1, MSRA2
+        Default: XAVIER
+    std : float, optional
+        Specifies the standard deviation value when the ``init`` parameter is set to NORMAL.
+    mean : float, optional
+        Specifies the mean value when the ``init`` parameter is set to NORMAL.
+    truncation_factor : float, optional
+        Specifies the truncation threshold (truncationFactor x std), when the ``init`` parameter is set to NORMAL
+    init_bias : float, optional
+        Specifies the initial bias for the layer.
+    dropout : float, optional
+        Specifies the dropout rate.
+        Default: 0
+    include_bias : bool, optional
+        Includes bias neurons (default).
+    src_layers : iter-of-Layers, optional
+        Specifies the layers directed to this layer.
+
+    Returns
+    -------
+    :class: 'GroupConv2d`
+
+    '''
+    type = 'groupconvo'
+    type_label = 'GroupConvo.'
+    type_desc = 'Group convolution layer'
+    can_be_last_layer = False
+    number_of_instances = 0
+
+    def __init__(self, n_filters, n_groups = 1, width = None, height = None, stride = 1, name = None,
+                 stride_horizontal = None, stride_vertical = None, padding = None, padding_width = None,
+                 padding_height = None, act = 'relu', fcmp_act = None, init = None, std = None, mean = None,
+                 truncation_factor = None, init_bias = None, dropout = None, include_bias = True,
+                 src_layers = None, **kwargs):
+        super(GroupConv2d, self).__init__(n_filters=n_filters, width=width,
+                                          height=height, stride=stride, name=name, stride_horizontal=stride_horizontal,
+                                          stride_vertical=stride_vertical, padding=padding, padding_width=padding_width,
+                                          padding_height = padding_height, act=act, fcmp_act=fcmp_act, init=init,
+                                          std=std, mean=mean, truncation_factor=truncation_factor, init_bias=init_bias,
+                                          dropout=dropout, include_bias=include_bias, src_layers=src_layers, **kwargs)
+        self.n_groups = n_groups
+        self.config['nGroups'] = n_groups
+
+    @property
+    def output_size(self):
+        if self._output_size is None:
+            # calculate output according to specified padding
+            # calculate output according to specified padding
+            if self.padding != (None, None):
+                out_h = (self.src_layers[0].output_size[0] - self.config['height'] + 2 * self.padding[0]) // \
+                        self.stride[0] + 1
+                out_w = (self.src_layers[0].output_size[1] - self.config['width'] + 2 * self.padding[1]) // self.stride[
+                    1] + 1
```

Comment:
This is a very strange line break.  If you don't want to have to use line-continuation characters, you can just surround the whole expression in parentheses.